core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,1347 @@
+            data: Some(zksync_basic_types::web3::Bytes(data)),
+            ..Default::default()
+        };
+
+        let res = self
+            .signing_client
+            .as_ref()
+            .call_contract_function(
+                call_request,
+                Some(BlockId::Number(BlockNumber::Number(context_block.into()))),
+            )
+            .await
+            .map_err(ServiceManagerError::EnrichedClient)?;
+
+        Ok(res.0.to_vec())
+    }
+
+    /// Verifies the certificate batch hash
+    pub(crate) async fn verify_batch(&self, blob_info: &BlobInfo) -> Result<(), VerificationError> {
+        let expected_hash = self.call_batch_id_to_metadata_hash(blob_info).await?;
+
+        if expected_hash == vec![0u8; 32] {
+            return Err(VerificationError::EmptyHash);
+        }
+
+        let actual_hash = self.hash_batch_metadata(
+            &blob_info
+                .blob_verification_proof
+                .batch_medatada
+                .batch_header,
+            &blob_info
+                .blob_verification_proof
+                .batch_medatada
+                .signatory_record_hash,
+            blob_info
+                .blob_verification_proof
+                .batch_medatada
+                .confirmation_block_number,
+        );
+
+        if expected_hash != actual_hash {
+            return Err(VerificationError::DifferentHashes {
+                expected: hex::encode(&expected_hash),
+                actual: hex::encode(&actual_hash),
+            });
+        }
+        Ok(())
+    }
+
+    fn decode_bytes(&self, encoded: Vec<u8>) -> Result<Vec<u8>, VerificationError> {
+        let output_type = [ParamType::Bytes];
+        let tokens: Vec<Token> = ethabi::decode(&output_type, &encoded)
+            .map_err(|e| ServiceManagerError::Decoding(e.to_string()))?;
+        let token = tokens
+            .first()
+            .ok_or(ServiceManagerError::Decoding("No tokens found".to_string()))?;
+        match token {
+            Token::Bytes(data) => Ok(data.to_vec()),
+            _ => Err(VerificationError::from(ServiceManagerError::Decoding(
+                "Token type mismatch".to_string(),
+            ))),
+        }
+    }
+
+    async fn get_quorum_adversary_threshold(
+        &self,
+        quorum_number: u32,
+    ) -> Result<u8, VerificationError> {
+        let func_selector = ethabi::short_signature("quorumAdversaryThresholdPercentages", &[]);
+        let data = func_selector.to_vec();
+
+        let call_request = CallRequest {
+            to: Some(self.cfg.svc_manager_addr),
+            data: Some(zksync_basic_types::web3::Bytes(data)),
+            ..Default::default()
+        };
+
+        let res = self
+            .signing_client
+            .as_ref()
+            .call_contract_function(call_request, None)
+            .await
+            .map_err(ServiceManagerError::EnrichedClient)?;
+
+        let percentages = self.decode_bytes(res.0.to_vec())?;
```

Comment:
Nit: `to_vec()` is redundant; `res.0` is already `Vec<u8>`.

Reply:
Fixed here https://github.com/lambdaclass/zksync-era/pull/386